include/sys/util.h

```diff
@@ -965,11 +967,12 @@ u8_t u8_to_dec(char *buf, u8_t buflen, u8_t value);
  *     int a6;
  *
  * @param F Macro takes nth variable argument as the argument.
+ * @param sep Separator (e.g. comma or semicolon). Must be in brackets.
```

Comment:
"Parentheses".  In common usage "brackets" often means "[]".  

Also this requirement (which I'm sure is for some very good reason down in the preprocessor layer) is... well... **just absolute voodoo insanity** to a typical user.  I think it deserves to be called out explicitly with its own paragraph or @note or whatever in the documentation for the macro.  e.g. "For technical reasons, the separator expression MUST be enclosed in parentheses.  These parentheses will be removed from the token stream in the final output, but must appear in the macro expansion at the top level."

Frankly I don't even know that that is correct.  But those are the questions I'd want answered before I touched a tool like this. :)

Reply:
Thanks for the comment. C preprocessor and voodoo have much in common.

Explanation is simple: preprocessor treats each comma as argument separator and parentheses (not brackets) are the only way to tell preprocessor to not look inside. I was considering adding something like `UTIL_COMMA` `UTIL_SEMICOLON` but that may create even more confusion.

I will add more explicit note.

Reply:
I support @andyross 's comment here and I agree with this PR being a good idea. I ran into a similar problem myself with DT_INST_FOREACH_STATUS_OKAY which @galak fixed in https://github.com/zephyrproject-rtos/zephyr/pull/25113. I think this style also makes it easier to write macros which do not anger checkpatch.

Reply:
> I will add more explicit note.

done